write-fonts/src/graph.rs

```diff
@@ -900,6 +1025,66 @@ mod tests {
         }
     }
 
+    #[test]
+    fn split_overflowing_spaces() {
+        // this attempts to show a simplified version of a gsub table with extension
+        // subtables, before any isolation/deduplication has happened.
+        //
+        //    before                         after
+        //      0           (GSUB)             0
+        //      |                              |
+        //      1        (lookup List)         1
+        //      |                              |
+        //      2          (Lookup)            2
+        //     / \                            / \
+        //  ╔═3   4═╗   (ext subtables)    ╔═3   4═╗
+        //  ║       ║                      ║       ║   (long offsets)
+        //  5─┐   ┌─6    (subtables)       5       6
+        //  │ └─8─┘ │                     / \     / \
+        //  │       │    (cov tables)    7'  8'  7   8
+        //  └───7───┘
+        //
```

Comment:
Diagram appreciated, without it the test would be inscrutable

Reply:
Agreed, I found it impossible to come up with good test cases without drawing them out first.